lib/nerves/package/providers/docker.ex

```diff
@@ -389,15 +396,53 @@ defmodule Nerves.Package.Providers.Docker do
   defp container_stop(name) do
     cmd = "docker"
     args = ["stop", name]
-    case System.cmd(cmd, args) do
+    case System.cmd(cmd, args, stderr_to_stdout: true) do
       {_, 0} -> :ok
+      {<<"Error response from daemon: ", response :: binary>>, _} ->
+
+        if response =~ "No such container" do
+          :ok
+        else
+          Mix.raise """
+          Nerves Docker provider could not stop container #{name}
+          Reason: #{response}
+          """
+        end
       {error, _} -> Mix.raise """
         The Nerves Docker provider could not stop Docker container #{name}
         Reason: #{error}
         """
     end
   end
 
+  defp container_delete(nil), do: :noop
+  defp container_delete(name) do
+    container_stop(name)
+    cmd = "docker"
+    args = ["rm", name]
+    case System.cmd(cmd, args, stderr_to_stdout: true) do
+      {_, 0} ->
+        :ok
+      {<<"Error response from daemon: ", response :: binary>>, _} ->
```

Comment:
Part of me is worried these messages will change and break our stuff, but I don't know of a better way to handle this.  There's similarity measures `String.jaro_distance` but I don't feel great about that either.

```elixir
iex(13)> String.jaro_distance("Error response from daemon:", "Error response from daemon was:")
0.956989247311828
```

We'll just have to keep an eye out for this.